config/config-server/src/com/thoughtworks/go/config/MagicalGoConfigXmlWriter.java

```diff
@@ -88,7 +89,19 @@ public void write(CruiseConfig configForEdit, OutputStream output, boolean skipP
         LOGGER.debug("[Serializing Config] Finished writing config partial.");
     }
 
-    private void verifyXsdValid(Document document) throws Exception {
+    public Document documentFrom(CruiseConfig config) {
+        Document document = createEmptyCruiseConfigDocument();
+        write(config, document.getRootElement(), configCache, registry);
+        return document;
+    }
+
+    public String toXml(Document document) throws IOException {
+        org.apache.commons.io.output.ByteArrayOutputStream outputStream = new org.apache.commons.io.output.ByteArrayOutputStream();
+        XmlUtils.writeXml(document, outputStream);
+        return outputStream.toString(Charset.defaultCharset());
```

Comment:
Default to `utf-8` via `StandardCharsets.UTF_8` instead of `Charset.defaultCharset()`